app/helpers/stream_entries_helper.rb

```diff
@@ -48,22 +48,25 @@ def microformats_h_class(status, is_predecessor, is_successor, include_threads)
   end
 
   def rtl?(text)
-    rtl_characters = /[\p{Hebrew}|\p{Arabic}|\p{Syriac}|\p{Thaana}|\p{Nko}]+/m.match(text)
+    rtl_charcount = text.gsub(/[^\p{Hebrew}|\p{Arabic}|\p{Syriac}|\p{Thaana}|\p{Nko}]/m, '').strip.size.to_f
 
-    if rtl_characters.present?
-      total_size = text.strip.size.to_f
-      rtl_size(rtl_characters.to_a) / total_size > 0.3
+    if rtl_charcount.positive?
+      # Remove mentions before counting characters to decide RTL ratio
+      justtext = text.gsub(Account::MENTION_RE, '')
+      # Remove Email addresses
+      justtext = justtext.gsub(/^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,6})$/, '')
+      # Naiive catcher for URLs
+      justtext = justtext.gsub(/[-A-Za-z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-A-Za-z0-9@:%_\+.~#?&\/=]*)/, '')
```

Comment:
Check https://github.com/tootsuite/mastodon/blob/master/app/lib/language_detector.rb#L32 for how URLs are stripped out when detecting language. Actually, the language detector should probably strip Account::MENTION_RE too. I don't think regexing e-mail addresses out is worth it, e-mail regex is ridiculously complex when done correctly.

Also, I think Ruby could also just rely on detected language instead of counting utf8 chars.